pandas/tests/extension/test_arrow.py

```diff
@@ -1125,6 +1125,38 @@ def test_comp_masked_numpy(self, masked_dtype, comparison_op):
         expected = pd.Series(exp, dtype=ArrowDtype(pa.bool_()))
         tm.assert_series_equal(result, expected)
 
+    def test_groupby_agg_extension(self, data_for_grouping):
+        # GH#38980 groupby agg on extension type fails for non-numeric types
+        df = pd.DataFrame({"A": [1, 1, 2, 2, 3, 3, 1, 4], "B": data_for_grouping})
+
+        def expected_inferred_result_dtype(dtype):
+            pa_dtype = dtype.pyarrow_dtype
+            if pa.types.is_date(pa_dtype):
+                return "date32[day][pyarrow]"
+            elif pa.types.is_time(pa_dtype):
+                return "time64[us][pyarrow]"
+            elif pa.types.is_decimal(pa_dtype):
+                return ArrowDtype(pa.decimal128(4, 3))
```

Comment:
On closer look, I think this is a bug being introduced here. This test is using `.first()`, it should be preserving the dtype in all cases. The changes in this PR now ignore the `preserve_dtype` argument of `agg_series`. When that is true, we should be passing `same_dtype=True` to `maybe_cast_pointwise_result`.

Reply:
resolved, keeping checks for if result is struct, for cases where lambda returns an object and preserve_dtype = false